Src/EB/AMReX_EB2_2D_C.cpp

```diff
@@ -20,87 +21,96 @@ void set_eb_data (const int i, const int j,
+            vcent(i,j,0,1) = (0.125_rt*dayp*dx[1]*dx[1] + ny*dx[0]*0.5_rt*bcent(i,j,0,1)*bcent(i,j,0,1)) / (vfrac(i,j,0) + 1.e-30_rt);
         }
     } else if (nyabs < tiny || nxabs > almostone) {
-        vfrac(i,j,0) = 0.5_rt*(aym+ayp);
-        if (vfrac(i,j,0) > almostone) {
+        vfrac(i,j,0) = 0.5_rt*(aym+ayp)*dx[1];
+        if (vfrac(i,j,0)/(dx[0]*dx[1]) > almostone) {
             vcent(i,j,0,0) = 0.0_rt;
         } else {
-            vcent(i,j,0,0) = (0.125_rt*(axp-axm) + nx*0.5_rt*bcent(i,j,0,0)*bcent(i,j,0,0)) / (vfrac(i,j,0) + 1.e-30_rt);
+            vcent(i,j,0,0) = (0.125_rt*daxp*dx[0]*dx[0] + nx*0.5_rt*bcent(i,j,0,0)*bcent(i,j,0,0)) / (vfrac(i,j,0) + 1.e-30_rt);
         }
         vcent(i,j,0,1) = 0.0_rt;
     } else {
-        Real aa = nxabs/ny;
-        const Real dx = x_ym - x_yp;
-        const Real dx2 = dx * (x_ym + x_yp);
-        const Real dx3 = dx * (x_ym*x_ym + x_ym*x_yp + x_yp*x_yp);
-        const Real af1 = 0.5_rt*(axm+axp) + aa*0.5_rt*dx2;
-        vcent(i,j,0,0) = 0.125_rt*(axp-axm) + aa*(1._rt/6._rt)*dx3;
-
-        aa = nyabs/nx;
+        Real aa = nxabs/ny*dx[0]/dx[1];
+        const Real dxx = x_ym - x_yp;
+        const Real dx2 = dxx * (x_ym + x_yp);
+        const Real dx3 = dxx * (x_ym*x_ym + x_ym*x_yp + x_yp*x_yp);
+        const Real af1 = 0.5_rt*(axm+axp)*dx[0] + aa*0.5_rt*dx2;
+        vcent(i,j,0,0) = -0.125_rt*daxp*dx[0]*dx[0] + aa*(1._rt/6._rt)*dx3;
+
+        aa = nyabs/nx*dx[1]/dx[1];
         const Real dy = y_xm - y_xp;
         const Real dy2 = dy * (y_xm + y_xp);
         const Real dy3 = dy * (y_xm*y_xm + y_xm*y_xp + y_xp*y_xp);
-        const Real af2 = 0.5_rt*(aym+ayp) + aa*0.5_rt*dy2;
-        vcent(i,j,0,1) = 0.125_rt*(ayp-aym) + aa*(1._rt/6._rt)*dy3;
+        const Real af2 = 0.5_rt*(aym+ayp)*dx[1] + aa*0.5_rt*dy2;
+        vcent(i,j,0,1) = -0.125_rt*dayp*dx[1]*dx[1] + aa*(1._rt/6._rt)*dy3;
 
         vfrac(i,j,0) = 0.5_rt*(af1+af2);
-        if (vfrac(i,j,0) > 1.0_rt-small) {
-            vfrac(i,j,0) = 1.0_rt;
+        if (vfrac(i,j,0)/(dx[0]*dx[1]) > 1.0_rt-small) {
+            vfrac(i,j,0) = dx[0]*dx[1];
             vcent(i,j,0,0) = 0.0_rt;
             vcent(i,j,0,1) = 0.0_rt;
-        } else if (vfrac(i,j,0) < small) {
+        } else if (vfrac(i,j,0)/(dx[0]*dx[1]) < small) {
             vfrac(i,j,0) = 0.0_rt;
             vcent(i,j,0,0) = 0.0_rt;
             vcent(i,j,0,1) = 0.0_rt;
         } else {
             vcent(i,j,0,0) *= (1.0_rt/vfrac(i,j,0));
             vcent(i,j,0,1) *= (1.0_rt/vfrac(i,j,0));
-            vcent(i,j,0,0) = amrex::min(amrex::max(vcent(i,j,0,0),Real(-0.5)),Real(0.5));
-            vcent(i,j,0,1) = amrex::min(amrex::max(vcent(i,j,0,1),Real(-0.5)),Real(0.5));
+            vcent(i,j,0,0) = amrex::min(amrex::max(vcent(i,j,0,0),Real(-0.5)*dx[0]),Real(0.5)*dx[0]);
+            vcent(i,j,0,1) = amrex::min(amrex::max(vcent(i,j,0,1),Real(-0.5)*dx[1]),Real(0.5)*dx[1]);
```

Comment:
Maybe we should move the clamping after vcent is divided by dx to avoid roundoff errors.

Reply:
Done.